poc/tenant-controller/pkg/controllers/tenants/controller.go

```diff
@@ -28,13 +30,18 @@ import (
 
 // Controller is k8s controller managing Tenant CRDs.
 type Controller struct {
-	informer cache.SharedIndexInformer
+	informer      cache.SharedIndexInformer
+	tenantsclient tenantsclient.Interface
+	k8sclient     k8sclient.Interface
+	tenantsList   map[string]tenantsapi.Tenant
```

Comment:
Regarding `tenantsList`, we will eventually need this map. However I don't see the need at this point.

Reply:
For the value type of the map, it's better to use "*tenantsapi.Tenant" to avoid duplicating the object, as we only read from these objects.

Reply:
I agree its not critical to have tenantsList right now.  One use to have it right now is that when somene does a tenant delete, we do not need to rely on the namespace objects provided in the manifest for the delete call. We do not want to have any namespaces getting leaked if the user did not list the correct set of namespaces for the tenant during the deletion, hence ideally its safer to save the state in the tenantsList object. This makes it easier for the user to not specify the namespace list when deleting a tenant and also protects against errors if they provide an incorrect list of namespaces during the tenant deletion.  Let me know if you still prefer that we remove the tenantsList or you feel its better to keep it ? In future we can also use it to check against namespace duplication across tenants and so on.

Reply:
Also note that when we do namespace auto-allocation, we will anyway not have the namespace list in the obj.